docs/DeveloperGuide.md

```diff
@@ -1,35 +1,306 @@
+
+Step 3: The user now then decides that adding this task was a mistake, and decides to remove the task from the
+To-Do List. The user does this by inputting the command `delete 1` into the terminal to delete a task in the task list.
+The command will then call the `TaskList#deleteTask()`, which removes the task at id 1 of the TaskList.
+
+The following sequence diagram shows how the delete task operation works:
+
+![DeleteTaskCommandSequence](images/DeleteTaskCommandSequence.png)
+
+Step 4: The user then decides to execute the command list. This command does not modify the TaskList.
+Thus, the TaskList will return to its initial state where there are no tasks stored in the TaskList.
+
+### Mark/unmark task feature
+
+Step 1: The user launches the program for the first time. The ToDoListManager will be initialised. This in turn will
+then initialise the Parser, TaskList and Storage. Take it as there are no existing tasks read/stored by the program.
+
+Step 2: The user executes `add survey -due 20/03/2023 23:59` command to add a task for the To-Do List.
+The add command calls `TaskList#addTask()`, which causes a new Task to be added to the existing TaskList.
+
+Step 3: The user wants to mark the task as completed by inputting the command `mark 1` into the terminal
+to mark the task as done. The command will then call the `TaskList#setDone`, which marks the task at id 1
+of the TaskList as done.
+
+For the unmark command, the user can instead input the command `unmark 1` to set the task as incomplete.
+The command also calls `TaskList#setDone` which sets the task at id 1 to be not done.
+
+The following sequence diagram shows how the mark/unmark task operation works:
+
+![MarkOrUnmarkTaskCommandSequence](images/MarkorUnmarkTaskCommandSequence.png)
+
+### Edit task deadline feature
+
+The edit deadline function extends NUS To-Do List with an edit feature for the deadlines assigned to tasks.
+It is facilitated by the TaskList and Command classes. It implements the `TaskList#editDeadline()` operation,
+which edits deadline of task at assigned id.
+
+
+Given below is an example usage scenario and how the edit deadline mechanism will behave at each step.
+
+Step 1. The user launches the application for the first time. There are no existing tasks read by the program.
+
+Step 2. The user executes `add survey -due 20/03/2023 23:59` command to add a task to the To-Do List.
+The `add` command calls `TaskList#addTask()`, which causes a new Task to be added to the existing TaskList.
+
+Step 3. The user has received an update about an extension to the deadline for the task, and decides to change
+the deadline by executing the `edit 1 -due 25/03/2023 15:00` command. The `edit` will call `TaskList#setDeadline()`,
+which updates the value of deadline for the Task item saved at id 1 to the new updated deadline.
+
+The following sequence diagram shows how the edit operations works:
+
+![EditDeadlineCommandSequence](images/EditDeadlineCommandSequence.png)
```

Comment:
There seems to be missing activation bars. 
![image](https://user-images.githubusercontent.com/88194809/228704799-906b6851-d809-4fa0-890b-6e9b00019d59.png)
I think there a missing "(" for .FLAG()